File: src/main/java/br/com/sicredi/api/service/CpfService.java
package br.com.sicredi.api.service;

import br.com.sicredi.api.client.CpfVoteClient;
import br.com.sicredi.api.domain.Cpf;
import br.com.sicredi.api.domain.enu.EligibleToVote;
import br.com.sicredi.api.dto.response.CpfExternalResponse;
import br.com.sicredi.api.exception.InvalidCpfException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class CpfService {

    private final Logger logger = LoggerFactory.getLogger(CpfService.class);

    @Autowired
    private CpfVoteClient cpfVoteClient;

    public Cpf getCpfIsValid(String cpf) {
        CpfExternalResponse info = cpfVoteClient.getCpfEnableToVote(cpf);
        String status = info.getStatus();
        boolean isAble = status.equals(EligibleToVote.ABLE_TO_VOTE.name());
        return new Cpf(cpf, isAble, status);
    }

    public void validateCpf(String cpf) {
        logger.info("Validating CPF {}", cpf);
        this.cpfOk(cpf);
    }

    private void cpfOk(String cpf) {
        if (!getCpfIsValid(cpf).getAbleVote()) {
            throw new InvalidCpfException(cpf);
        }
    }
}

File: validate-random-cpf/src/main/kotlin/br/com/sicredi/generate_cpf/service/CpfService.kt
package br.com.sicredi.generate_cpf.service

import br.com.sicredi.generate_cpf.exception.InvalidCpfException
import br.com.sicredi.generate_cpf.model.Status
import br.com.sicredi.generate_cpf.model.enu.StatusEnum
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import kotlin.random.Random

@Service
class CpfService {

    val log: Logger = LoggerFactory.getLogger(CpfService::class.java)

    fun generateCpfStatus(cpf: String): Status {
        if (!this.validateCpf(cpf)) {
            throw InvalidCpfException("Cpf is not valid").also { log.error("Invalid CPF") }
        }

        val textStatus = if (Random.nextBoolean()) StatusEnum.ABLE_TO_VOTE.name else StatusEnum.UNABLE_TO_VOTE.name
        return Status(textStatus).also { log.info("CPF is valid") }
    }

    private fun validateCpf(document: String): Boolean {
        if (document.isEmpty()) return false

        val numbers = document.filter { it.isDigit() }.map {
            it.toString().toInt()
        }

        if (numbers.size != 11) return false

        if (numbers.all { it == numbers[0] }) return false

        val dv1 = ((0..8).sumOf { (it + 1) * numbers[it] }).rem(11).let {
            if (it >= 10) 0 else it
        }

        val dv2 = ((0..8).sumOf { it * numbers[it] }.let { (it + (dv1 * 9)).rem(11) }).let {
            if (it >= 10) 0 else it
        }

        return numbers[9] == dv1 && numbers[10] == dv2
    }
}
